WooCommerce/WooCommerceTests/Yosemite/InAppPurchaseStoreTests.swift

```diff
@@ -176,35 +176,35 @@ final class InAppPurchaseStoreTests: XCTestCase {
     }
 
     func test_user_is_entitled_to_product_returns_false_when_not_entitled() throws {
-        // Given
-
         // When
-        let result = waitFor { promise in
-            let action = InAppPurchaseAction.userIsEntitledToProduct(productID: self.sampleProductID) { result in
-                    promise(result)
-                }
-            self.store.onAction(action)
+        Task {
+            let isEntitled = try await userIsEntitledToProduct()
+            // Then
+            XCTAssertFalse(try isEntitled.get())
         }
-
-        // Then
-        let isEntitled = try XCTUnwrap(result.get())
-        XCTAssertFalse(isEntitled)
     }
 
     func test_user_is_entitled_to_product_returns_true_when_entitled() throws {
         // Given
         try storeKitSession.buyProduct(productIdentifier: sampleProductID)
 
         // When
-        let result = waitFor { promise in
+        Task {
+            let isEntitled = try await userIsEntitledToProduct()
+            // Then
+            XCTAssertTrue( try isEntitled.get())
+        }
+    }
+}
+
+private extension InAppPurchaseStoreTests {
+    // Returns whether the user is entitled the product identified with the passed ID
+    func userIsEntitledToProduct() async throws -> Result<Bool, Error> {
```

Comment:
maybe we can test with `InAppPurchasesForWPComPlansManager` which does a similar interface conversion?